mongo/integration/unified/event_verification.go

```diff
@@ -405,3 +437,145 @@ func stringifyEventsForClient(client *clientEntity) string {
 
 	return str.String()
 }
+
+func getNextServerDescriptionChangedEvent(
+	events []*event.ServerDescriptionChangedEvent,
+) (*event.ServerDescriptionChangedEvent, []*event.ServerDescriptionChangedEvent, error) {
+	if len(events) == 0 {
+		return nil, nil, errors.New("no server changed event published")
+	}
+
+	return events[0], events[1:], nil
+}
+
+func getNextServerHeartbeatStartedEvent(
+	events []*event.ServerHeartbeatStartedEvent,
+) (*event.ServerHeartbeatStartedEvent, []*event.ServerHeartbeatStartedEvent, error) {
+	if len(events) == 0 {
+		return nil, nil, errors.New("no heartbeat started event published")
+	}
+
+	return events[0], events[1:], nil
+}
+
+func getNextServerHeartbeatSucceededEvent(
+	events []*event.ServerHeartbeatSucceededEvent,
+) (*event.ServerHeartbeatSucceededEvent, []*event.ServerHeartbeatSucceededEvent, error) {
+	if len(events) == 0 {
+		return nil, nil, errors.New("no heartbeat succeeded event published")
+	}
+
+	return events[0], events[:1], nil
+}
+
+func getNextServerHeartbeatFailedEvent(
+	events []*event.ServerHeartbeatFailedEvent,
+) (*event.ServerHeartbeatFailedEvent, []*event.ServerHeartbeatFailedEvent, error) {
+	if len(events) == 0 {
+		return nil, nil, errors.New("no heartbeat failed event published")
+	}
+
+	return events[0], events[:1], nil
+}
+
+func getNextTopologyDescriptionChangedEvent(
+	events []*event.TopologyDescriptionChangedEvent,
+) (*event.TopologyDescriptionChangedEvent, []*event.TopologyDescriptionChangedEvent, error) {
+	if len(events) == 0 {
+		return nil, nil, errors.New("no topology description changed event published")
+	}
+
+	return events[0], events[:1], nil
+}
+
+func verifySDAMEvents(client *clientEntity, expectedEvents *expectedEvents) error {
```

Comment:
@qingyang-hu @matthewdale The addition of the 1.17 schema in this ticket and the UTR updates missed in the implementation of https://github.com/ShaneHarvey/specifications/commit/d6b8cce6abb3b8e1a0b8f1dc7ee737e18322cfce require this verification function.